pythonPackages update: asn1ate and asn2quickder

leenaars opened the pull request:
###### Motivation for this change

Upgrade to latest released version of these asn1 tools.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

leenaars committed:
arpa2cm: init -> 0.5

leenaars committed:
pythonPackages.asn1ate: unstable -> 0.6

dotlambda commented on pkgs/development/python-modules/asn1ate/default.nix:
```diff
@@ -3,13 +3,12 @@
-  date = "20160810";
-  version = "unstable-${date}";
+  version= "0.6";
   name = "${pname}-${version}";
```
You can delete the `name`.

leenaars replied:
Isn't that mandatory?

dotlambda replied:
Not for Python. `buildPythonPackage` will automatically derive ot from `pname` and `version`.

leenaars replied:
Done. Thanks for helping out!

dotlambda commented on pkgs/development/tools/asn2quickder/default.nix:
```diff
@@ -1,30 +1,22 @@
-stdenv.mkDerivation rec {
+python2Packages.buildPythonPackage rec {
   pname = "asn2quickder";
   name = "${pname}-${version}";
```
Leave out name here as well.

leenaars replied:
Done.

dotlambda commented on pkgs/development/tools/asn2quickder/default.nix:
```diff
@@ -1,30 +1,22 @@
+{ stdenv, fetchFromGitHub, python2Packages, makeWrapper, cmake }:
 
-stdenv.mkDerivation rec {
+python2Packages.buildPythonPackage rec {
```
Better use `python2Packages.callPackage` and take `buildPythonPackage` as well as all dependencies as arguments.

leenaars replied:
I don't understand what you mean by that? I thought his was a very common way of doing this.
Do you have a recommended example where this is done elsewhere in nixpkgs.
I grepped but could not find something that seems to fit here ...

dotlambda replied:
In `all-packages.nix`, one can use `python2Packages.callPackage`. And in `default.nix` the first line should look something like
```
{ stdenv, fetchFromGitHub, python2Packages, makeWrapper, cmake, pytestrunner, pytest, six, pyparsing }:
```

dotlambda replied:
Actually, this should be `buildPythonApplication`, since it's not a library.

leenaars replied:
@dotlambda: When I do that, for some reason it tries to create a wheel and cannot find six. So it doesn't build anymore.

It builds perfectly fine as is. Would it not be okay to leave it as is? I only did a version bump...

leenaars replied:
Thanks. Done.

leenaars commented:
@dotlambda: I removed the 'name' but it turns out then you get a real ugly prefix, e.g. python2.7-asn1ate-0.6 and python2.7-asn2quickder-1.2-5 instead of asn1ate-0.6. So I reverted back to what I had.

dotlambda commented:
Actually, I think this should be called `quick-der` now.
We can add `asn2quicker` as an alias.
But then, it doesn't make sense to use `buildPythonApplication`, since `quick-der` is much more than a python executable.
I don't have enough experience to know how to package C and Python libraries at the same time. Maybe @FRidh can help.

leenaars commented:
I don't know either, my solution was to keep them apart. That at least I understand and works... I have a separate quickder on my system which I will submit soon.

I don't think this should cost anyone a lot of time at this stage. When we would run into any issues later on, this could be fixed then. This is very useful but not mainstream technical tooling...

leenaars commented:
All suggestions have been processed, should be ready to merge.

Mic92 commented:
@GrahamcOfBorg build quickder

GrahamcOfBorg reviewed (commented):
Failure for system: x86_64-linux

```
Skipped ASN.1 header file generation for rfc2578.asn1
rm -f quick-der
ln -s . quick-der
Skipped compilation of skipped header files
install: cannot stat 'rfc3280.h': No such file or directory
make[1]: *** [Makefile:39: install] Error 1
make[1]: Leaving directory '/tmp/nix-build-quickder-1.0-RC2.drv-0/source/rfc'
make: *** [Makefile:20: install] Error 2
builder for ‘/nix/store/c82bfpjpiizfmxg90qymz9gkm5xb4rkl-quickder-1.0-RC2.drv’ failed with exit code 2
error: build of ‘/nix/store/c82bfpjpiizfmxg90qymz9gkm5xb4rkl-quickder-1.0-RC2.drv’ failed
```

GrahamcOfBorg reviewed (commented):
Failure for system: x86_64-darwin

```
Package ‘quickder-1.0-RC2’ in /tmp/nix-ofborg/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/lnl7-mac/pkgs/development/libraries/quickder/default.nix:32 is not supported on ‘x86_64-darwin’, refusing to evaluate.

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowBroken = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowBroken = true; }
to ~/.config/nixpkgs/config.nix.

```

GrahamcOfBorg reviewed (commented):
Failure for system: aarch64-linux

```
Skipped ASN.1 header file generation for rfc2578.asn1
rm -f quick-der
ln -s . quick-der
Skipped compilation of skipped header files
install: cannot stat 'rfc3280.h': No such file or directory
make[1]: *** [Makefile:39: install] Error 1
make[1]: Leaving directory '/build/source/rfc'
make: *** [Makefile:20: install] Error 2
builder for '/nix/store/aqy0wjncd0f4g5qzzjh2iwqqhjrxw3nm-quickder-1.0-RC2.drv' failed with exit code 2
error: build of '/nix/store/aqy0wjncd0f4g5qzzjh2iwqqhjrxw3nm-quickder-1.0-RC2.drv' failed
```

Mic92 commented:
Was `quickder` broken before?

FRidh commented:
`asn2quickder` builds on master but not with this commit. Some fixing is needed.

leenaars committed:
asn2quickder: 0.7RC1 -> 1.2-6

leenaars committed:
quickder: 1.0-RC2 -> 1.2-6

leenaars commented:
I updated quickder to the latest release, should work now.

FRidh commented:
@GrahamcOfBorg build quickder

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.34161&attempt_id=95129920-a0ac-43c1-a517-f0a96780904d)

<details><summary>Partial log (click to expand)</summary><p>

```
make[1]: *** Waiting for unfinished jobs....
[ 97%] Built target rfc4531_asn1_h
[ 98%] Build Python script rfc5805.py from ASN.1 spec
[ 98%] Built target rfc4120_asn1_h
[ 98%] Built target rfc5805_asn1_h
[ 98%] Built target kxover_asn1_h
[ 98%] Built target RemotePKCS11_asn1_h
make: *** [Makefile:152: all] Error 2
builder for '/nix/store/8amsviz8pz2gjyd6mcad0k3wa62c4dpp-quickder-1.2-6.drv' failed with exit code 2
error: build of '/nix/store/8amsviz8pz2gjyd6mcad0k3wa62c4dpp-quickder-1.2-6.drv' failed
```
</p></details>

FRidh commented:
@leenaars quickder still seems to be failing.

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.34161&attempt_id=b79788b1-139b-4c84-bdde-7001a50331d2)

<details><summary>Partial log (click to expand)</summary><p>

```
[100%] Built target rfc5805_asn1_h
[100%] Built target rfc4120_asn1_h
[100%] Built target rfc4373_asn1_h
[100%] Built target rfc2315_asn1_h
[100%] Built target rfc7292_asn1_h
[100%] Built target kxover_asn1_h
[100%] Built target RemotePKCS11_asn1_h
make: *** [Makefile:152: all] Error 2
builder for '/nix/store/gwnyic3yjxqpz85bm572vcajgri02jzz-quickder-1.2-6.drv' failed with exit code 2
error: build of '/nix/store/gwnyic3yjxqpz85bm572vcajgri02jzz-quickder-1.2-6.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.34161&attempt_id=5f411adf-87ad-4486-b040-ad08e6323e47)

<details><summary>Partial log (click to expand)</summary><p>

```
Package ‘quickder-1.2-6’ in /private/var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/lnl7-mac/pkgs/development/libraries/quickder/default.nix:51 is not supported on ‘x86_64-darwin’, refusing to evaluate.

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowBroken = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowBroken = true; }
to ~/.config/nixpkgs/config.nix.

```
</p></details>

leenaars commented:
It builds on my NixOS and Debian systems. The failure is in both cases a script that is called which fails on:
/usr/bin/env: bad interpreter: No such file or directory

A little digging reveals that a chrooted system does not have /usr/bin/env by design.
(https://github.com/NixOS/nixpkgs/issues/6227)...

I used patchShebangs to hopefully fix the issue.

leenaars commented:
@GrahamcOfBorg build quickder

Mic92 commented:
@GrahamcOfBorg build quickder

GrahamcOfBorg commented:
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.34161&attempt_id=abeb816d-6d8f-464e-971c-3e9d958b64c3)

<details><summary>Partial log (click to expand)</summary><p>

```
Package ‘quickder-1.2-6’ in /private/var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/lnl7-mac/pkgs/development/libraries/quickder/default.nix:51 is not supported on ‘x86_64-darwin’, refusing to evaluate.

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowBroken = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowBroken = true; }
to ~/.config/nixpkgs/config.nix.

```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.34161&attempt_id=58ae67f5-bd27-462a-baf1-a8f325bcb38f)

<details><summary>Partial log (click to expand)</summary><p>

```
Scanning dependencies of target asn1-spec-modules
[ 97%] Built target asn1-spec-modules
Scanning dependencies of target kxover_asn1_md
[ 98%] Build markdown text file kxover.md from ASN.1 spec
/nix/store/fcxliihmhm2ak9z4890gk0qw63zsbrcx-bash-4.4-p12/bin/bash: /tmp/nix-build-quickder-1.2-6.drv-0/source/python/scripts/asn1literate: /usr/bin/env: bad interpreter: No such file or directory
make[2]: *** [arpa2/CMakeFiles/kxover_asn1_md.dir/build.make:61: arpa2/doc/kxover.md] Error 126
make[1]: *** [CMakeFiles/Makefile2:1167: arpa2/CMakeFiles/kxover_asn1_md.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
builder for ‘/nix/store/np8fkjzrp1lay1avgsw0a0yyclrzk61l-quickder-1.2-6.drv’ failed with exit code 2
error: build of ‘/nix/store/np8fkjzrp1lay1avgsw0a0yyclrzk61l-quickder-1.2-6.drv’ failed
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.34161&attempt_id=50310202-a4c5-4bd2-9f7d-5daf9a036958)

<details><summary>Partial log (click to expand)</summary><p>

```
[100%] Built target rfc4531_asn1_h
[100%] Built target rfc4120_asn1_h
[100%] Built target rfc4373_asn1_h
[100%] Built target rfc2315_asn1_h
[100%] Built target rfc7292_asn1_h
[100%] Built target kxover_asn1_h
[100%] Built target RemotePKCS11_asn1_h
make: *** [Makefile:152: all] Error 2
builder for '/nix/store/10lbn5321ihfqcjzgwbrn6zyflf84l7j-quickder-1.2-6.drv' failed with exit code 2
[31;1merror:[0m build of '/nix/store/10lbn5321ihfqcjzgwbrn6zyflf84l7j-quickder-1.2-6.drv' failed
```
</p></details>

Mic92 committed:
quickder: fix python shebangs

Mic92 commented:
@GrahamcOfBorg build quickder

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.34161&attempt_id=b30926db-a060-4fda-8b5d-2540c413c4ef)

<details><summary>Partial log (click to expand)</summary><p>

```
-- Installing: /nix/store/hhc46r9wnmz16hshafgqmf6b11ynap4k-quickder-1.2-6/share/doc/quick-der/kxover.md
-- Installing: /nix/store/hhc46r9wnmz16hshafgqmf6b11ynap4k-quickder-1.2-6/share/doc/quick-der/RemotePKCS11.md
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/hhc46r9wnmz16hshafgqmf6b11ynap4k-quickder-1.2-6
shrinking /nix/store/hhc46r9wnmz16hshafgqmf6b11ynap4k-quickder-1.2-6/lib/libquickder.so
strip is /nix/store/adidfx4pa7vmvby0gjqqmiwg2x49yr27-binutils-2.28.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/hhc46r9wnmz16hshafgqmf6b11ynap4k-quickder-1.2-6/lib 
patching script interpreter paths in /nix/store/hhc46r9wnmz16hshafgqmf6b11ynap4k-quickder-1.2-6
checking for references to /tmp/nix-build-quickder-1.2-6.drv-0 in /nix/store/hhc46r9wnmz16hshafgqmf6b11ynap4k-quickder-1.2-6...
/nix/store/hhc46r9wnmz16hshafgqmf6b11ynap4k-quickder-1.2-6
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.34161&attempt_id=b7a9d8cb-d76c-41e6-8b5a-acc414a9e7e8)

<details><summary>Partial log (click to expand)</summary><p>

```
-- Installing: /nix/store/jc4vv1dkhnlljnn00gmqlil55gbj3xz2-quickder-1.2-6/share/doc/quick-der/kxover.md
-- Installing: /nix/store/jc4vv1dkhnlljnn00gmqlil55gbj3xz2-quickder-1.2-6/share/doc/quick-der/RemotePKCS11.md
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/jc4vv1dkhnlljnn00gmqlil55gbj3xz2-quickder-1.2-6
shrinking /nix/store/jc4vv1dkhnlljnn00gmqlil55gbj3xz2-quickder-1.2-6/lib/libquickder.so
strip is /nix/store/skd6ix5ipkyhxzq7naylj4digawakl4j-binutils-2.28.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/jc4vv1dkhnlljnn00gmqlil55gbj3xz2-quickder-1.2-6/lib
patching script interpreter paths in /nix/store/jc4vv1dkhnlljnn00gmqlil55gbj3xz2-quickder-1.2-6
checking for references to /build in /nix/store/jc4vv1dkhnlljnn00gmqlil55gbj3xz2-quickder-1.2-6...
/nix/store/jc4vv1dkhnlljnn00gmqlil55gbj3xz2-quickder-1.2-6
```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.34161&attempt_id=007e52fa-e997-4b3d-a773-2571693f153c)

<details><summary>Partial log (click to expand)</summary><p>

```
Package ‘quickder-1.2-6’ in /private/var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/lnl7-mac/pkgs/development/libraries/quickder/default.nix:53 is not supported on ‘x86_64-darwin’, refusing to evaluate.

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowBroken = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowBroken = true; }
to ~/.config/nixpkgs/config.nix.

```
</p></details>
